quantity-pickers/src/main/java/com/decathlon/vitamin/compose/quantity/pickers/VitaminQuantityPickers.kt

```diff
@@ -69,59 +75,98 @@ object VitaminQuantityPickers {
-                        } else {
-                            modifier.width(156.dp)
-                        }
+            ) {
+                Row(
+                    modifier = Modifier
+                        .height(48.dp)
+                        .then(
+                            if (isExpanded) {
+                                modifier.fillMaxWidth()
+                            } else {
+                                modifier.width(156.dp)
+                            }
+                        )
+                ) {
+                    IconButton(
+                        painter = painterResource(id = R.drawable.ic_vtmn_subtract_fill),
+                        contentDescription = stringResource(id = R.string.vtmn_subtract_button_description),
+                        enabled = subtractEnabled,
+                        colors = colors,
+                        shape = shape.copy(
+                            topEnd = CornerSize(0.dp),
+                            bottomEnd = CornerSize(0.dp)
+                        ),
+                        onClick = onSubtractClicked
                     )
+
+                    MiddleTextField(
+                        modifier = Modifier
+                            .background(
+                                color = colors.backgroundColor.copy(
+                                    alpha = if (!addEnabled && !subtractEnabled) {
+                                        VtmnStatesDisabled
+                                    } else VtmnStatesEnabled
+                                )
+                            )
+                            .fillMaxHeight()
+                            .weight(1.0f),
+                        addEnabled = addEnabled,
+                        subtractEnabled = subtractEnabled,
+                        colors = colors,
+                        textStyle = textStyle,
+                        value = value,
+                        onValueChange = onValueChange,
+                        editTextEnabled = editTextEnabled,
+                        keyboardOptions = keyboardOptions,
+                        keyboardActions = keyboardActions
+                    )
+
+                    IconButton(
+                        painter = painterResource(id = R.drawable.ic_vtmn_add_fill),
+                        contentDescription = stringResource(id = R.string.vtmn_add_button_description),
+                        enabled = addEnabled,
+                        colors = colors,
+                        shape = shape.copy(
+                            topStart = CornerSize(0.dp),
+                            bottomStart = CornerSize(0.dp)
+                        ),
+                        onClick = onAddClicked
+                    )
+                }
+                ErrorLabel(errorMessage = errorMessage)
+            }
+        }
+    }
+
+    @Composable
+    private fun ErrorLabel(
+        errorMessage: String?
+    ) {
+        errorMessage?.let {
```

Comment:
It is weird to have this check here and do thing if `errorMessage` is `null`.
Maybe you can make this check in the `Primary` component and convert `errorMessage` to no nullable parameter?